src/main/java/traincraft/blocks/assemblytables/TileAssemblyTable.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Traincraft
+ * Copyright (c) 2011-2020.
+ *
+ * This file ("TileAssemblyTable.java") is part of the Traincraft mod for Minecraft.
+ * It is created by all people that are listed with @author below.
+ * It is distributed under LGPL-v3.0.
+ * You can find the source code at https://github.com/Traincraft/Traincraft
+ */
+package traincraft.blocks.assemblytables;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.entity.item.EntityItem;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.InventoryBasic;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+import traincraft.tile.BaseTile;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * This is the Assembly Table tileEntity for all tiers of assembly table. It gets the tier based on where the player is looking.
+ *
+ * @implNote If we can figure out how to register all the assembly table tile entities as one, we don't need the three extra files as wrappers for this.
+ *
+ * @author PseudonymPatel
+ * @since 2020-8-7
+ */
+public class TileAssemblyTable extends BaseTile {
+
+    private final int tier;
+    private GuiAssemblyTable guiAssemblyTable;
+    private ContainerAssemblyTable containerAssemblyTable;
```

Comment:
This is not eh behaviour any minecraft tile entity has. Container and Guis are created when a player opens the gui and destroyed when it is closed. They should never be stored

Reply:
I had to do this because I had to call a detectAndSendChanges in the container, otherwise items would never show up in the output slots. I agree that is not the way to do it, but I couldn't figure out an alternative. Any suggestions? The train workbench works in an entirely different way, because it uses the 3x3 matrix and a buncha vanilla stuff, so I can't really take inspiration from there.